skt/__init__.py

```diff
@@ -518,24 +539,73 @@ def mktgz(self, clean=True):
 
         logging.info("building kernel: %s", args)
 
-        mk = subprocess.Popen(args,
-                              stdout=subprocess.PIPE,
-                              stderr=subprocess.STDOUT)
-        (stdout, stderr) = mk.communicate()
-        for line in stdout.split("\n"):
-            m = re.match("^Tarball successfully created in (.*)$", line)
-            if m:
-                tgzpath = m.group(1)
-                break
+        with io.open(self.buildlog, 'wb') as writer, \
+                io.open(self.buildlog, 'rb') as reader:
+            make = subprocess.Popen(args,
+                                    stdout=writer,
+                                    stderr=subprocess.STDOUT)
+            make.timedout = False
```

Comment:
Hmm, is it normal to just go and add a field to an instance of some 3rd-party class? What if a later version of subprocess adds a field named the same, but used for something else?

Reply:
You're right! It is not a good idea. Anyway, I was looking at the code again and... The race still there, because if the process ends after the poll call, it will report a timeout error when the process finished correctly... So I think I'm going back to the first implementation which it is simpler. What do you think?

Reply:
Well, if the process was still running at the moment of the poll call, it means it timed out. I.e. it took longer than allowed. So, I think that part is OK.

Reply:
Fair enough, please take a look at it again.

Reply:
OTOH, previously you simply checked if the timer has expired or not, and that might have happened after make has finished. E.g. because of the sleep in the loop, or because your output to stdout was blocked.